collect_app/src/androidTest/java/org/odk/collect/android/feature/formentry/IntentGroupTest.java

```diff
@@ -80,17 +95,109 @@ public void appMissingErrorText_ShouldComeFromSpecialFormText() {
     }
 
     @Test
-    public void externalApp_ShouldPopulateFields() {
+    public void externalApp_ShouldPopulateStringFields() {
         Intent resultIntent = new Intent();
-        resultIntent.putExtra("sometext", "Here is a text value");
 
-        int randomInteger = (new Random()).nextInt(255);
-        resultIntent.putExtra("someinteger", randomInteger);
+        int randomInteger = new Random().nextInt(255);
+        double scale = Math.pow(10, 2);
+        double randomDecimal = Math.round(new Random().nextDouble() * scale) / scale;
+        String randomText = RandomString.make();
```

Comment:
Do these values need to be random or are you just using random values to generate test data?

Reply:
It was just for test data it's not required and I got rid of it to simplify the long `externalApp_ShouldPopulateFields` test.